src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/HeaderVersionRule.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base.Deployments;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    public abstract class HeaderVersionRule : ConsensusRule
```

Comment:
Brilliant I think this is a good approach.

Reply:
Ah please add documentation on this class on what is its intent.
>An base skeleton that is implemented by networks to define and verify the version of blocks.

Reply:
Done